hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -72,8 +73,8 @@
   private final int maxRetriesMultiplier;
   private final boolean eofAutoRecovery;
 
-  //Indicates whether this particular worker is running
-  private boolean isReaderRunning = true;
+  // Indicates whether this particular worker is running
+  private final AtomicBoolean isReaderRunning = new AtomicBoolean(true);
```

Comment:
A volatile boolean is enough?

Reply:
The volatile keyword is a fickle thing and this is not time-sensitive code.  More straight forward to use the Java concurrent package